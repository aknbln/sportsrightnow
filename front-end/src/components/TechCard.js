import React from "react";

import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";

const TechCard = (props) => {
  const { title, image, text, url } = props.techData;
  return (
    <Card
    style= {{backgroundColor: 'lightgray'}}>
      <Card.Img variant="top" src={image} style={{width: '100%', height: '20vw', objectFit: 'cover'}}/>
      <Card.Body>
        <Card.Title style = {{fontSize: '4vh', color: '#404040'}}>{title}</Card.Title>
        <Card.Text style = {{fontSize: '2.5vh', color: '#404040'}}>{text}</Card.Text>
      </Card.Body>
      <Card.Footer style= {{backgroundColor: 'dimgray'}}>
        <Button href={url} class="btn btn-primary stretched-link"
        style= {{backgroundColor: '#3d405b', borderColor: 'black'}}>
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default TechCard;
