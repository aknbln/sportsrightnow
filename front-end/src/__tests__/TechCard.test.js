import renderer from 'react-test-renderer'
import TechCard from '../components/TechCard'
import img from '../assets/images/question.png'

const Jest = 
{
    title: "Jest",
    image: img,
    text: "Javascript unit test library",
    url: "https://jestjs.io/",
}

it('Tech Card link correct', () => {
    const component = renderer.create(<TechCard techData={Jest}/>)
    expect(component.url = "https://jestjs.io/")
})

it('DevCard renders correctly', () => {
    const component = renderer.create(<TechCard techData={Jest}/>)
    let json = component.toJSON()
    expect(json).toMatchSnapshot()
})