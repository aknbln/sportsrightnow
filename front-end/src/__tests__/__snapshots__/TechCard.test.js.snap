// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  <img
    className="card-img-top"
    src="question.png"
    style={
      Object {
        "height": "20vw",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      style={
        Object {
          "color": "#404040",
          "fontSize": "4vh",
        }
      }
    >
      Jest
    </div>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2.5vh",
        }
      }
    >
      Javascript unit test library
    </p>
  </div>
  <div
    className="card-footer"
    style={
      Object {
        "backgroundColor": "dimgray",
      }
    }
  >
    <a
      class="btn btn-primary stretched-link"
      className="btn btn-primary"
      disabled={false}
      href="https://jestjs.io/"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "#3d405b",
          "borderColor": "black",
        }
      }
      tabIndex={0}
    >
      More Info
    </a>
  </div>
</div>
`;
